rsp_jupyter_extensions/handlers/query.py

```diff
@@ -255,15 +277,21 @@ def _get_query_text_list(self, job_ids: list[str]) -> list[TAPQuery]:
         self.log.info(f"Requesting query history for {job_ids}")
         for job in job_ids:
             try:
-                retval.append(self._get_query_text_job(job))
+                retval.append(await self._get_query_text_job(job))
             except Exception:
                 self.log.exception(f"job {job} text retrieval failed")
         return retval
 
-    def _get_query_text_job(self, job: str) -> TAPQuery:
+    async def _get_query_text_job(self, job: str) -> TAPQuery:
         if job in self._cache:
             return TAPQuery(jobref=job, text=self._cache[job])
-        resp = self._tap_client.get(f"async/{job}")
+        # If there is no colon, it's whatever is behind /api/tap.
+        if job.find(":") == -1:
+            resp = await self._tap_client.get(f"async/{job}")
+        else:
+            # We have a dataset, and presumably a matching client.
+            ds, job_id = job.split(":")
+            resp = await self._dataset_client[ds].get(f"async/{job_id}")
```

Comment:
Do you think we should we check that the dataset exists in the client and handle that error explicitly?

Reply:
Sure, no harm in it.